lib/katello/tasks/update_subscription_facet_backend_data.rake

```diff
@@ -0,0 +1,61 @@
+namespace :katello do
+  task :update_subscription_facet_backend_data => ["environment"] do
+    def error(message)
+      @errors << message
+      @errors << "\n"
+    end
+
+    def report_errors
+      if @errors.any?
+        filename = "subscription_facet_upgrade-#{Time.now.to_i}.log"
+        path = "/var/log/foreman/#{filename}"
+        path = "/tmp/#{filename}" unless File.writable?(path)
+
+        file = File.open(path, 'w')
+        @errors.each { |error| file.write(error) }
+        file.close
+        $stderr.print "***********************************\n"
+        $stderr.print "*************WARNING***************\n"
+        $stderr.print "Errors detected during upgrade step.\n"
+        $stderr.print "Details saved to: #{file.path}\n"
+        $stderr.print "This step can be rerun with:\n"
+        $stderr.print "  foreman-rake katello:update_subscription_facet_backend_data\n"
+        $stderr.print "You are likely encountering a bug.\n"
+        $stderr.print "***********************************\n"
+      end
+    end
+
+    @errors ||= []
+    User.current = User.anonymous_api_admin
+    puts _("Updating backend data for subscription facets")
+
+    #there may be some invalid hosts, if there are create a primary interface
+    ::Host.includes(:interfaces).find_each do |host|
+      if host.primary_interface.nil?
+        host.interfaces.create!(:primary => true)
```

Comment:
How this happened in the first place? Primary interface should exist for every host now. Creating it this way might fail because of some interfaces validation added later.

Core migration that should have taken care of it
* https://github.com/theforeman/foreman/blob/develop/db/migrate/20140910153654_move_host_nics_to_interfaces.rb
* https://github.com/theforeman/foreman/blob/develop/db/migrate/20150508124600_copy_unmanaged_hosts_to_interfaces.rb